src/lazy/expanded/compiler.rs

```diff
@@ -1098,30 +1098,12 @@ impl TemplateCompiler {
         Ok(())
     }
 
-    fn resolve_maybe_macro_id_expr<D: Decoder>(
-        tdl_context: TdlContext<'_>,
-        id_expr: Option<IonResult<LazyValue<'_, D>>>,
-    ) -> IonResult<Arc<Macro>> {
-        // Get the name or address from the `Option<IonResult<LazyValue<_>>>` if possible, or
-        // surface an appropriate error message.
-        let value = match id_expr {
-            None => {
-                return IonResult::decoding_error(
-                    "found an empty s-expression in an unquoted context",
-                )
-            }
-            Some(Err(e)) => return Err(e),
-            Some(Ok(value)) => value,
-        };
-        Self::resolve_macro_id_expr(tdl_context, value)
-    }
-
     /// Given a `LazyValue` that represents a macro ID (name or address), attempts to resolve the
     /// ID to a macro reference.
     fn resolve_macro_id_expr<D: Decoder>(
         tdl_context: TdlContext<'_>,
         id_expr: LazyValue<'_, D>,
-    ) -> IonResult<Arc<Macro>> {
+    ) -> IonResult<Option<Arc<Macro>>> {
```

Comment:
🪧 This now returns `Ok(None)` if reading the expression works but no macro is found with the resulting ID. This allows it to be used in places where not finding a macro ID is a legal possibility.